### PR TITLE
feat: Implement S3 export temporal workflow

### PR DESCRIPTION
**Pull request by tomasfarias**
## Problem

Work in progress for Temporal exports.

## Changes

👉 *Stay up-to-date with [PostHog coding conventions](https://posthog.com/docs/contribute/coding-conventions) for a smoother review.*

### CODE REVIEW

File: posthog/models/export/export.py [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+    unpaused_at: models.DateTimeField = models.DateTimeField(null=True)
+    start_at: models.DateTimeField = models.DateTimeField(null=True)
+    end_at: models.DateTimeField = models.DateTimeField(null=True)
+    name: models.CharField = models.CharField()
+    destination: models.ForeignKey = models.ForeignKey("ExportDestination", on_delete=models.CASCADE)
+    calendars: ArrayField = ArrayField(models.JSONField(), default=list)
+    intervals: ArrayField = ArrayField(models.JSONField(), default=list)
+    cron_expressions: ArrayField = ArrayField(models.CharField(), default=list)
+    skip: ArrayField = ArrayField(models.JSONField(), default=list)
+    jitter: models.DurationField = models.DurationField(null=True)
```
Comment (lharries) [inline posthog/models/export/export.py:106]: what does jitter refer to?

File: posthog/models/export/export.py [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+        S3 = "S3"
+
+    destinations_to_workflows = {
+        Destination.S3: (S3ExportWorkflow, S3ExportInputs),
+    }
+    type: models.CharField = models.CharField(choices=Destination.choices)
+    team: models.ForeignKey = models.ForeignKey("Team", on_delete=models.CASCADE)
+    created_at: models.DateTimeField = models.DateTimeField(auto_now_add=True)
+    last_updated_at: models.DateTimeField = models.DateTimeField(auto_now_add=True)
+    parameters: models.JSONField = models.JSONField(default=dict, blank=True)
```
[the remark this answers is not in the document]
Comment (lharries) [reply posthog/models/export/export.py:31]: will switch the frontend to config

[the remark this answers is not in the document]
Comment (tomasfarias) [reply posthog/models/export/export.py:31]: Cool, I'll push a change.

File: frontend/src/scenes/cdp/ExportOverview.tsx [hunk 3]
```diff
@@ -0,0 +1,277 @@
...
+export function StatusToTagType(status: BatchExportStatus): LemonTagPropsType {
+    switch (status) {
+        case BatchExportStatus.Running:
+            return 'highlight'
+        case BatchExportStatus.Cancelled:
+            return 'warning'
+        case BatchExportStatus.Completed:
+            return 'success'
+        case BatchExportStatus.ContinuedAsNew:
+            return 'highlight' // TODO: understand what this does and choose a more appropriate color
```
Comment (tomasfarias) [inline frontend/src/scenes/cdp/ExportOverview.tsx:36]: In a few words, `Continued-As-New` means a run was retried. In more words: A new instance of a workflow (a workflow run) is created by Temporal with the same state as an existing instance. This new workflow run has it's own, different id and history.

File: frontend/src/lib/api.ts [hunk 4]
```diff
@@ -561,6 +582,75 @@ const api = {
...
+                runId: BatchExportRunType['id'],
+                teamId: TeamType['id'] = getCurrentTeamId()
+            ): Promise<BatchExportRunType> {
+                return await new ApiRequest().batchExportRun(runId, teamId).get()
+            },
+            async create(
+                data: Partial<BatchExportRunType>,
+                teamId: TeamType['id'] = getCurrentTeamId()
+            ): Promise<BatchExportRunType> {
+                return await new ApiRequest().batchExportRuns(teamId).create({ data })
```
Comment (tomasfarias) [inline frontend/src/lib/api.ts:628]: @lharries `ExportRuns` cannot be created by API.

**Comment by tomasfarias** [comment]
This PR was created to track development and collaborate. For this reason, and for the purposes of reducing scope, I will be dividing the work into Frontend and Backend and closing this PR 

The latter can already be found here: https://github.com/PostHog/posthog/pull/15364

**Pull request closed without merging**